File: src/main/java/nnglebanov/daoexample/repositories/CommentRepository.java
package nnglebanov.daoexample.repositories;

import nnglebanov.daoexample.domain.Comment;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.transaction.annotation.Transactional;

@Transactional
public interface CommentRepository extends JpaRepository<Comment, Integer> {
}

File: src/main/java/nnglebanov/daoexample/repositories/CommentRepository.kt
package nnglebanov.daoexample.repositories

import nnglebanov.daoexample.domain.Comment
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import org.springframework.stereotype.Repository
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

@Repository
interface ReactiveCommentRepository : ReactiveMongoRepository<Comment, String> {
    override fun findAll(): Flux<Comment>

    override fun findById(id: String): Mono<Comment>

    fun save(person: Mono<Comment>): Mono<Comment>
}